server/models/post.model.js

```diff
@@ -76,6 +78,41 @@ const PostSchema = new mongoose.Schema({
  * Methods
  */
 PostSchema.method({
+  updateVote (vote, user) {
+    vote.upVote();
+
+    const userIdString = user._id.toString();
+    const postIdString = this._id.toString();
+    if (vote.active) {
+      this.score += incrementValue;
+      raccoon.liked(userIdString, postIdString);
```

Comment:
Will look deeper at this a bit later but  some quick feedback. My thinking was that  voting would be inherited (in this example updateVote & upVote) from a base model that would implement this functionality and both Comment and Post extend that base model.  Since post is a little unique, it would basically call super.updateVote() which would return whether this was a `liked` or `unliked` and then would make a call to raccoon. Alternatively the base model could have blank methods called liked() , unliked(), etc and then the Post would overwrite those.

Reply:
That's exactly what I sent you. This is the correct way to move towards that approach.